src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbol.cs

```diff
@@ -703,6 +704,13 @@ internal override void AfterAddingTypeMembersChecks(ConversionsBase conversions,
                 Debug.Assert(explicitInterfaceSpecifier != null);
                 _explicitInterfaceType.CheckAllConstraints(conversions, new SourceLocation(explicitInterfaceSpecifier.Name), diagnostics);
             }
+
+            if (_refKind == RefKind.RefReadOnly)
+            {
+                DeclaringCompilation.EnsureIsReadOnlyAttributeExists(diagnostics, _location, modifyCompilationForRefReadOnly: true);
```

Comment:
We have tests covering these diagnostics, like `IsReadOnlyAttributeExistsWithWrongConstructorSignature_Assembly`. However, Covering every permutation for every symbol type and error type and location (parameter vs return type) I believe should be left to the final test plan (lot's of moving pieces right now in this feature).
However, I've added 5 more tests for symbols for that specific error location for this PR.